sw/device/lib/testing/dif_uart_test.cc

```diff
@@ -0,0 +1,819 @@
+    dif_uart_enable_t state = kDifUartDisable;
+    EXPECT_READ32(UART_INTR_STATE_REG_OFFSET, irq.second);
+
+    EXPECT_TRUE(dif_uart_irq_state_get(&dif_uart_, irq.first, &state));
+
+    EXPECT_EQ(state, kDifUartEnable);
+  }
+
+  for (auto const& irq : uart_irq_reg_map_) {
+    dif_uart_enable_t state = kDifUartEnable;
+    EXPECT_READ32(UART_INTR_STATE_REG_OFFSET, ~irq.second);
+
+    EXPECT_TRUE(dif_uart_irq_state_get(&dif_uart_, irq.first, &state));
+
+    EXPECT_EQ(state, kDifUartDisable);
+  }
+
+  EXPECT_EQ(uart_irq_reg_map_.size(), 8);
+}
+
+// dif_uart_irq_state_clear() tests
+
+/**
+ * TODO
+ */
+TEST_F(DifUartTestUartInterrupt, IrqStateClearFailureUartNull) {
+  EXPECT_FALSE(dif_uart_irq_state_clear(nullptr, kDifUartInterruptTxWatermark));
+}
+
+/**
+ * Tests all combinations of dif_uart_irq_state_clear().
+ * TODO !!!add ...Count values to dif_uart enums!!!
+ */
+TEST_F(DifUartTestUartInterrupt, IrqStateClearSuccess) {
+  for (auto const& irq : uart_irq_reg_map_) {
+    EXPECT_READ32(UART_INTR_STATE_REG_OFFSET,
+                  0b00000000'00000000'00000000'00000000);
+    EXPECT_WRITE32(UART_INTR_STATE_REG_OFFSET, irq.second);
+    EXPECT_TRUE(dif_uart_irq_state_clear(&dif_uart_, irq.first));
+  }
+
+  EXPECT_EQ(uart_irq_reg_map_.size(), 8); // TODO, sould be separate test
+}
+
+// dif_uart_irqs_disable() tests
+
+/**
+ * TODO
+ */
+TEST_F(DifUartTestUartInitialised, IrqsDisableFailureUartNull) {
+  uint32_t state;
+  EXPECT_FALSE(dif_uart_irqs_disable(nullptr, &state));
+}
+
+/**
+ * TODO
+ */
+TEST_F(DifUartTestUartInitialised, IrqsDisableSuccess) {
+  EXPECT_WRITE32(UART_INTR_ENABLE_REG_OFFSET,
+                 0b00000000'00000000'00000000'00000000);
+
+  EXPECT_TRUE(dif_uart_irqs_disable(&dif_uart_, nullptr));
+}
+
+/**
+ * TODO
+ */
+TEST_F(DifUartTestUartInitialised, IrqsDisableStateSuccess) {
+  // IRQs disabled
+  EXPECT_READ32(UART_INTR_ENABLE_REG_OFFSET,
+                0b00000000'00000000'00000000'00000000);
+  EXPECT_WRITE32(UART_INTR_ENABLE_REG_OFFSET,
+                 0b00000000'00000000'00000000'00000000);
+
+  uint32_t state = 0b11111111'11111111'11111111'11111111;
+  EXPECT_TRUE(dif_uart_irqs_disable(&dif_uart_, &state));
+
+  EXPECT_EQ(state, 0b00000000'00000000'00000000'00000000);
+
+  // IRQs enabled
+  EXPECT_READ32(UART_INTR_ENABLE_REG_OFFSET,
+                0b11111111'11111111'11111111'11111111);
+  EXPECT_WRITE32(UART_INTR_ENABLE_REG_OFFSET,
+                 0b00000000'00000000'00000000'00000000);
+
+  state = 0b00000000'00000000'00000000'00000000;
+  EXPECT_TRUE(dif_uart_irqs_disable(&dif_uart_, &state));
+
+  EXPECT_EQ(state, 0b11111111'11111111'11111111'11111111);
+}
+
+// dif_uart_irqs_restore() tests
+
+/**
+ * TODO
+ */
+TEST_F(DifUartTestUartInitialised, IrqsRestoreFailureUartNull) {
+  EXPECT_FALSE(dif_uart_irqs_restore(nullptr, 0xffffffff));
+}
+
+/**
+ * TODO
+ */
+TEST_F(DifUartTestUartInitialised, IrqsRestoreSuccess) {
+  uint32_t state = 0b11111111'11111111'11111111'11111111;
```

Comment:
You should inline this. If you want to refer to all ones or all zeroes I see no reason to not use `-1` or `0`.

Reply:
I have changed all ones to `0xffffffff` and all zeros to `0`. And I use `constexpr` for `0xffffffff` instead of hardcoding them.